tests/components/abode/test_config_flow.py

```diff
@@ -0,0 +1,80 @@
+"""Tests for the Abode config flow."""
+from unittest.mock import patch
+
+from homeassistant import data_entry_flow
+from homeassistant.components.abode import config_flow
+from homeassistant.components.abode.const import DOMAIN
+from homeassistant.const import CONF_PASSWORD, CONF_USERNAME
+
+from tests.common import MockConfigEntry
+
+
+async def test_show_form(hass):
+    """Test that the form is served with no input."""
+    flow = config_flow.AbodeFlowHandler()
+    flow.hass = hass
+
+    result = await flow.async_step_user(user_input=None)
+
+    assert result["type"] == data_entry_flow.RESULT_TYPE_FORM
+    assert result["step_id"] == "user"
+
+
+async def test_duplicate_error(hass):
+    """Test that errors are shown when duplicates are added."""
+    conf = {CONF_USERNAME: "user@email.com", CONF_PASSWORD: "password"}
+
+    MockConfigEntry(domain=DOMAIN, data=conf).add_to_hass(hass)
+    flow = config_flow.AbodeFlowHandler()
+    flow.hass = hass
+
+    result = await flow.async_step_user(user_input=conf)
+    assert result["errors"] == {CONF_USERNAME: "identifier_exists"}
+
+
+async def test_invalid_credentials(hass):
+    """Test that invalid credentials throws an error."""
+    from abodepy.exceptions import AbodeAuthenticationException
+
+    conf = {CONF_USERNAME: "user@email.com", CONF_PASSWORD: "password"}
+
+    flow = config_flow.AbodeFlowHandler()
+    flow.hass = hass
+
+    with patch("abodepy.Abode", side_effect=AbodeAuthenticationException("errors")):
```

Comment:
Didn't quite understand this but I had to add an argument to `AbodeAuthenticationException` for it to work. Does not make sense to me why it requires an argument.

Reply:
That exception requires a tuple as argument:
https://github.com/MisterWil/abodepy/blob/6f84bb428fd1da98855f55083cd427bebbcc57ae/abodepy/exceptions.py#L4-L17

https://github.com/MisterWil/abodepy/blob/61bb200ed21153a6d03e68044e9a729778f31d0a/abodepy/__init__.py#L139-L140

Reply:
I'm an idiot - that is very obvious now, lol. If I'm understanding this correctly, it really doesn't matter what argument is passed since really all I need is the `AbodeAuthenticationException` to assert to test that the config flow is functioning properly. Doesn't look like abodepy is currently doing anything with the arguments that are passed anyways.

Reply:
The argument must support access by index, to avoid raising an error when creating the exception. After that it may not matter. I guess it depends on what we do when excepting the exception.